addOns/postman/src/main/java/org/zaproxy/addon/postman/PostmanParser.java

```diff
@@ -75,15 +96,254 @@ public void importFromUrl(final String url) throws IllegalArgumentException, IOE
+                                .append(
+                                        URLEncoder.encode(
+                                                data.getValue(), StandardCharsets.UTF_8.name()));
+                    } catch (UnsupportedEncodingException e) {
+                    }
+                }
+            }
+
+            bodyContent = urlencodedBodySB.toString();
+        } else if (mode.equals(Body.FORM_DATA)) {
+            if (body.getFormData() == null) {
+                return httpMessage;
+            }
+
+            String boundary = "----" + System.currentTimeMillis();
+
+            contentType = "multipart/form-data; boundary=" + boundary;
+
+            StringBuilder formDataBodySB = new StringBuilder();
+            for (FormData formData : body.getFormData()) {
+                if (!formData.isDisabled()) {
+                    formDataBodySB
+                            .append(generateMultiPartBody(formData, boundary))
+                            .append(HttpHeader.CRLF);
+                }
+            }
+
+            formDataBodySB.append("--").append(boundary).append("--").append(HttpHeader.CRLF);
+
+            bodyContent = formDataBodySB.toString();
+        } else if (mode.equals(Body.FILE)) {
+            if (body.getFile() == null) {
+                return httpMessage;
+            }
+
+            String src = body.getFile().getSrc();
+
+            contentType = getFileContentType(src);
+
+            try {
+                bodyContent = FileUtils.readFileToString(new File(src), StandardCharsets.UTF_8);
+            } catch (IOException e1) {
+            }
+        } else if (mode.equals(Body.GRAPHQL)) {
+            if (body.getGraphQl() == null) {
+                return httpMessage;
+            }
+
+            contentType = HttpHeader.JSON_CONTENT_TYPE;
+
+            GraphQl graphQlBody = body.getGraphQl();
+            String query = graphQlBody.getQuery().replaceAll("\r\n", "\\\\r\\\\n");
+            String variables = graphQlBody.getVariables().replaceAll("\\s", "");
+
+            bodyContent = String.format("{\"query\":\"%s\", \"variables\":%s}", query, variables);
+        }
+
+        if (!isContentTypeAlreadySet(request.getHeader())) {
+            httpMessage.getRequestHeader().setHeader(HttpHeader.CONTENT_TYPE, contentType);
+        }
+
+        httpMessage.getRequestBody().setBody(bodyContent.toString());
+        httpMessage.getRequestHeader().setContentLength(httpMessage.getRequestBody().length());
+
+        return httpMessage;
+    }
+
+    private static String generateMultiPartBody(FormData formData, String boundary) {
+        StringBuilder multipartDataSB = new StringBuilder();
+
+        multipartDataSB.append("--").append(boundary).append(HttpHeader.CRLF);
+        multipartDataSB
+                .append("Content-Disposition: form-data; name=\"")
+                .append(formData.getKey())
+                .append("\"");
+
+        if ("file".equals(formData.getType())) {
```

Comment:
Use `Body.FILE`?

Reply:
Keeping the previous order (i.e. `Body.FILE.equals(formData.getType())`) would be better, to handle the case when `getType()` returns null (that's not being being checked beforehand).